test/apps/sockaddr/sa_util.cc

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "sa_util.h"
+
+#include <sys/epoll.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <cstring>
+#include <climits>
+
+
+error::error(const std::string& message) : m_message(message) {
+}
+
+error::~error() throw() {
+}
+
+const char* error::what() const throw() {
+    return m_message.c_str();
+}
+
+sys_error::~sys_error() throw() {
+}
+
+sys_error::sys_error(const std::string& message, int errn) :
+    error(message + ": " + strerror(errn) + " (" + std::to_string(errn) + ")") {
+}
+
+file_desc::file_desc(int fd) : m_fd(fd) {
+}
+
+file_desc::~file_desc() {
+    int ret = ::close(m_fd);
```

Comment:
the destructor closes fd opened outside of the class. It's better to manage resources in single scope.

Reply:
not sure it's possible, since sometimes "accept" returns a file descriptor, then we want to transfer ownership to file_desc to make sure it's closed (even in case of exception)